src/nanotron/trainer.py

```diff
@@ -206,9 +206,9 @@ def __init__(
             self.consumed_train_samples = 0
 
         # Setup tensorboard write and log writers on output rank
-        self.logger_ranks = self.parallel_context.world_rank_matrix[
+        self.logger_ranks = self.parallel_context.get_global_rank(
```

Comment:
It would be cool if we could get a global rank like get_global_rank(pp_rank=..., tp_rank=...)